docs/sources/flow/reference/components/discovery.consulagent.md

```diff
@@ -0,0 +1,125 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/discovery.consulagent/
+title: discovery.consulagent
+---
+
+# discovery.consulagent
+
+`discovery.consulagent` allows you to retrieve scrape targets from [Consul's Agent API][].
+Only the services registered with the local agent running on the same host will be watched.
+This is suitable for very large Consul clusters for which using the Catalog API would be too slow or resource intensive.
+
+[Consul's Agent API]: https://developer.hashicorp.com/consul/api-docs/agent
+
+## Usage
+
+```river
+discovery.consulagent "LABEL" {
+  server = CONSUL_SERVER
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+| Name               | Type           | Description                                                                                                    | Default          | Required |
+| ------------------ | -------------- | -------------------------------------------------------------------------------------------------------------- | ---------------- | -------- |
+| `server`           | `string`       | Host and port of the Consul Agent API.                                                                         | `localhost:8500` | no       |
+| `token`            | `secret`       | Secret token used to access the Consul Agent API.                                                              |                  | no       |
+| `datacenter`       | `string`       | Datacenter in which the Consul Agent is configured to run. If not provided, the default is used.               |                  | no       |
```

Comment:
> If not provided, the default is used

What is the default here?

Reply:
according to the code if not set explicitly the datacenter is retrieved from the config of the consulagent and would represent the local node's datacenter. I could improve the doc here.

Reply:
Actually, I'm not even sure that it makes sense to have this field exposed here. Why would someone need to set the datacenter when we can retrieve it from the agent directly?

Reply:
> Why would someone need to set the datacenter when we can retrieve it from the agent directly?

Oh you mean the _Consul_ agent here. Yeah, since the upstream struct allows to configure this, I think we should expose this. It's also a less-surprising UX from the converters perspective.

>  I could improve the doc here.
Yeah, let's do that and say where the default comes from.